src/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -897,41 +897,20 @@ internal void ProcessHandshakeSuccess()
                 size   -
                 output - Encrypted bytes
         --*/
-        internal SecurityStatusPal Encrypt(byte[] buffer, int offset, int size, ref byte[] output, out int resultSize)
+        internal SecurityStatusPal Encrypt(ReadOnlyMemory<byte> buffer, ref byte[] output, out int resultSize)
         {
             if (NetEventSource.IsEnabled)
             {
-                NetEventSource.Enter(this, buffer, offset, size);
-                NetEventSource.DumpBuffer(this, buffer, 0, Math.Min(buffer.Length, 128));
+                NetEventSource.Enter(this, buffer, 0, buffer.Length);
+                int eventLength = Math.Min(buffer.Length, 128);
+                NetEventSource.DumpBuffer(this, buffer.Slice(0, eventLength).ToArray(), 0, eventLength);
```

Comment:
We should probably add a DumpBuffer overload that takes a Memory, and skips having the offset and count ints.  The body of DumpBuffer can then try to extract the existing array if there is one and dump that out, or use ToArray as a last resort.

Reply:
I will look at this.